src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs

```diff
@@ -21,15 +27,23 @@ internal interface IRemoteSourceGenerationService
     /// compare that to the prior generated documents it has to see if it can reuse those directly, or if it needs to
     /// remove any documents no longer around, add any new documents, or change the contents of any existing documents.
     /// </summary>
-    ValueTask<ImmutableArray<(SourceGeneratedDocumentIdentity documentIdentity, SourceGeneratedDocumentContentIdentity contentIdentity, DateTime generationDateTime)>> GetSourceGenerationInfoAsync(
+    /// <remarks>
+    /// Should only be called by the "RegularCompilationTracker", and should only return data from its view of the
+    /// world.  Not from the view of a "GeneratedFileReplacingCompilationTracker".
+    /// </remarks>
+    ValueTask<ImmutableArray<RegularCompilationTrackerSourceGenerationInfo>> GetRegularCompilationTrackerSourceGenerationInfoAsync(
         Checksum solutionChecksum, ProjectId projectId, CancellationToken cancellationToken);
```

Comment:
this helps fix the sublte bug i mentioned in the OP.  When teh host calls to OOP to get the generated docs, it is calling from its "regular compilation tracker" (not the "generated file replacing compilation tracker").  As such, the call should go through to that same tracker on the OOP side, not the outermost wrapped tracker like we were doing before.

Reply:
The method rename seems a little weird to me. I wonder if it would be a bit nicer from the caller's side to use the original method name, and pass in an enum indicating the type of compilation tracker that is to be used?

Reply:
Might be best to chat in person. Maybe a better way I could have phrased the thought is could CompilationTracker have an identifier that could be passed across to determine the tracker to use?